kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -122,6 +124,17 @@ public Transaction build(Engine engine) {
                 shouldUpdateMetadata = true;
                 metadata = metadata.withNewConfiguration(newProperties);
             }
+            Set<String> newWriterFeatures =
+                    protocol.extractAutomaticallyEnabledWriterFeatures(metadata);
+            if (!newWriterFeatures.isEmpty()) {
+                shouldUpdateProtocol = true;
```

Comment:
Add a `logger.info` message saying that the following features are automatically enabled based on the metadata.